js/form.js

```diff
@@ -0,0 +1,132 @@
+const cancelBtn = document.getElementById("cancel-form-btn");
+submitBtn.disabled = true;
+loadPage();
+
+//*****************EVENT LISTENERS*****************/
+//CANCEL button functionality
+cancelBtn.addEventListener("click", cancelEvent);
+function cancelEvent() {
+    set(ref(db, 'eventToEdit/'), { //when user clicks cancel - no events to edit so make sure eventToEdit is set to none 
+        id: "none",
+        name: "none",
+        description: "none",
+        date: "none"
+    });
+    window.location.href = "list.html";
+}
+document.getElementById("event-name").addEventListener("keyup", validateForm);
+document.getElementById("event-date").addEventListener("change", validateForm);
+document.getElementById("event-description").addEventListener("keyup", validateForm);
+
+//*****************FUNCTIONS*****************/
+//Here, check if the page is to add a new event or edit an event and based on that load event listeners
+function loadPage() {
+    get(child(ref(db), 'eventToEdit/')).then((snapshot) => {
+        if (snapshot.exists()) {
+            let name = `${snapshot.val().name}`;
+            let description = `${snapshot.val().description}`;
+            let date = `${snapshot.val().date}`;
+            let id = `${snapshot.val().id}`;
+
+            if (name == "none") {   //To add a new event
+                document.getElementById("heading1").innerHTML = "Create Event";
+                document.getElementById("submit-form-btn").value = "Add";
+                submitBtn.addEventListener("click", (event) => {
+                    event.preventDefault();
+                    let eventName = document.getElementById("event-name").value;
+                    let eventDescription = document.getElementById("event-description").value;
+                    let eventDate = document.getElementById("event-date").value;
+                    let eventID = eventName + Math.floor((Math.random() * 9999) + 1);
+                    eventName = eventName.charAt(0).toUpperCase() + eventName.substr(1).toLowerCase();//convert to title case
+                    set(ref(db, 'events/' + eventID),
+                        {
+                            id: eventID,
+                            name: eventName,
+                            description: eventDescription,
+                            date: eventDate
+                        });
+                    window.location.href = "list.html";
+                });
+            }
+            else {   //To edit an event
+                document.getElementById("heading1").innerHTML = "Edit Event";
+                document.getElementById("submit-form-btn").value = "Save";
+                document.getElementById("event-name").value = name;
+                document.getElementById("event-description").value = description;
+                document.getElementById("event-date").value = date;
+                submitBtn.addEventListener("click", (event) => {
+                    event.preventDefault();
```

Comment:
lets move this logic to a function called submitForm

Reply:
done